ui/app.py

```diff
@@ -349,3 +361,30 @@ def _rally_agents(self, callback: callable):
         for agent in self._agents.values():
             if agent:
                 callback(agent)
+
+    def _read_history_dump(self):
+        try:
+            with open(DEBUG_FILEPATH, mode="r", encoding="utf-8") as file:
+                file_buffer = file.read()
+        except FileNotFoundError:
+            return
+
+        try:
+            starting_layout_str, game_history_str = json.loads(file_buffer)
+            starting_layout = BoardLayout[starting_layout_str]
+            game_history = GameHistory.decode(game_history_str)
+            self._load_game_state(starting_layout, game_history)
+        except Exception:
+            Debug.log(traceback.format_exc(), DebugType.Warning)
+            Debug.log(f"WARNING: {DEBUG_FILEPATH} is corrupted, removing...", DebugType.Warning)
+            os.remove(DEBUG_FILEPATH)  # remove corrupted file
```

Comment:
Would be nice to have a timestamp system here so debug files don't get lost to the abyss